src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -42,6 +43,205 @@ def get_server_location(server_name, resource_group_name):
+    def _has_min_status(node, min_status):
+        # Returns true if the status is at least as high as the filter
+        return status_list.index(node.status) >= status_list.index(min_status)
+
+    def _filter_by_status(nodes):
+        # Returns a new list with the status filter applied. If a node is in lenient_nodes, then
+        # keep it even if the status filter is 'Default' and the node's status is only 'Available'.
+        return [n for n in nodes if _has_min_status(n, status) or (
+            _has_min_status(n, CapabilityStatus.available) and n in lenient_nodes
+        )]
+
+    # Filter server versions.
+    capabilities.supported_server_versions = (
+        _filter_by_status(capabilities.supported_server_versions))
+
+    for sv in capabilities.supported_server_versions:
+        # Filter editions.
+        sv.supported_editions = _filter_by_status(sv.supported_editions)
+
+        for e in sv.supported_editions:
+            # Filter service objectives.
+            e.supported_service_level_objectives = (
+                _filter_by_status(e.supported_service_level_objectives))
+
+            for slo in e.supported_service_level_objectives:
+                # Filter max sizes.
+                slo.supported_max_sizes = _filter_by_status(slo.supported_max_sizes)
+
+        # Filter elastic pool editions
+        sv.supported_elastic_pool_editions = _filter_by_status(sv.supported_elastic_pool_editions)
+
+        for e in sv.supported_elastic_pool_editions:
+            # Filter supported DTUs
+            e.supported_elastic_pool_dtus = _filter_by_status(e.supported_elastic_pool_dtus)
+
+            for pool_dtu in e.supported_elastic_pool_dtus:
+                # Filter per database max DTUs
+                pool_dtu.supported_per_database_max_dtus = \
+                    _filter_by_status(pool_dtu.supported_per_database_max_dtus)
+
+                for db_max_dtu in pool_dtu.supported_per_database_max_dtus:
+                    # Filter per database min DTUs
+                    db_max_dtu.supported_per_database_min_dtus = \
+                        _filter_by_status(db_max_dtu.supported_per_database_min_dtus)
+
+                # Filter per database max size
+                pool_dtu.supported_per_database_max_sizes = \
+                    _filter_by_status(pool_dtu.supported_per_database_max_sizes)
+
+
+def _capabilities_prune_empty(capabilities):
+    for sv in capabilities.supported_server_versions:
+        # Remove editions with no service objectives (due to filters)
+        sv.supported_editions = [e for e in sv.supported_editions
+                                 if len(e.supported_service_level_objectives) > 0]
+
+    # Remove server versions with no editions (due to filters)
+    capabilities.supported_server_versions = [sv for sv in capabilities.supported_server_versions
+                                              if len(sv.supported_editions) > 0]
+
+
+def _capabilities_prune_by_depth(capabilities, depth):
```

Comment:
This gives pylint error for nested blocks too deep. I could eliminate all the `else`s since they are technically not needed - the subsequent `for`s would be iterating over an empty list. However I think it's more obvious what's going on when the `else`s are there.